x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/changes/get_metric_changes.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { AggregationsAutoDateHistogramAggregation } from '@elastic/elasticsearch/lib/api/types';
+import type { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+import type { AggregateOfMap, ChangePointType } from '@kbn/es-types/src/search';
+import type { ObservabilityAIAssistantElasticsearchClient } from '../../clients/elasticsearch';
+
+type MetricType = 'min' | 'max' | 'sum' | 'count' | 'avg' | 'p95' | 'p99';
+
+function getMetricAggregation({ field, type }: { field?: string; type: MetricType }) {
+  if (type === 'count') {
+    return field
+      ? {
+          value_count: {
+            field,
+          },
+        }
+      : {
+          filter: {
+            match_all: {},
+          },
+          bucket_path: '_count',
+        };
+  }
+
+  if (!field) {
+    throw new Error(`Metric type ${type} needs a field to aggregate over`);
+  }
+
+  if (type === 'min' || 'max' || 'sum' || 'avg') {
+    return {
+      [type]: {
+        field,
+      },
+    } as Record<Exclude<MetricType, 'count' | 'p95' | 'p99'>, { field: string }>;
+  }
+
+  const percentile = `${type.split('p')[1]}.0`;
+
+  return {
+    percentiles: {
+      field,
+      percents: [Number(percentile)],
+      keyed: true,
+    },
+    bucket_path: percentile,
+  };
+}
+
+export async function getMetricChanges({
+  index,
+  filters,
+  groupBy,
+  field,
+  type,
+  client,
+  dateHistogram,
+}: {
+  index: string;
+  filters: QueryDslQueryContainer[];
+  groupBy: string[];
+  field?: string;
+  type: MetricType;
+  client: ObservabilityAIAssistantElasticsearchClient;
+  dateHistogram: AggregationsAutoDateHistogramAggregation;
+}) {
+  const metricAgg = getMetricAggregation({
+    type,
+    field,
+  });
+
+  const groupAgg = groupBy.length
+    ? groupBy.length === 1
+      ? { terms: { field: groupBy[0] } }
+      : {
+          multi_terms: {
+            terms: groupBy.map((groupingField) => ({ field: groupingField })),
+            size: 10,
+          },
+        }
+    : {
+        filters: {
+          filters: [
+            {
+              match_all: {},
+            },
+          ],
+        },
+      };
```

Comment:
nit: imo nested ternaries can be difficult to read. 

```suggestion
  const getGroupAgg = () => {
    if (groupBy.length === 0) {
      return {
        filters: {
          filters: [
            {
              match_all: {},
            },
          ],
        },
      };
    }

    if (groupBy.length === 1) {
      return { terms: { field: groupBy[0] } };
    }

    if (groupBy.length > 1) {
      return {
        multi_terms: {
          terms: groupBy.map((groupingField) => ({ field: groupingField })),
          size: 10,
        },
      };
    }
  };
```

Reply:
ack, will change, and I like that I don't have to add explicit types with your suggestion